src/signature.c

```diff
@@ -21,6 +21,8 @@ struct p11prov_sig_ctx {
     P11PROV_SESSION *session;
 
     CK_RSA_PKCS_PSS_PARAMS pss_params;
+
+    bool digest_fallback;
```

Comment:
last think after couple of reads of this PR, I am not sure about the naming of this variable. Its kind of fallback, but digest fallback sounds confusing/misleading, when this is basically flag saying "digest is calculated separately from the signature/verification operation" so my proposal would be "separate_digest". Or at least have on this line a comment describing what this variable/flag is doing.

Reply:
I couldn't find a better name indeed, I used digest fallback, because the fallback is about separately calculating the digest, I could have called it mechanism_fallback, because it is about handling an unsupported mechanism.

Reply:
See the fixup commit I added.

Reply:
After fixing the typo, looks good.